In [82]:
%reload_ext autoreload
%autoreload 2

# ` Question of Interest: ` How many COMP BS majors are there in total when combining the students in session 1 and session 2?

First I will read in the csv and convert it to a column-oriented data table.

In [83]:
from data_utils import read_csv_rows, columnar

data_rows: list[dict[str, str]] = read_csv_rows('../../data/survey.csv')
columns: dict[str, list[str]] = columnar(data_rows)

{'': '0', 'row_number': '0', 'year': '21', 'unc_status': 'Returning UNC Student', 'comp_major': 'No', 'primary_major': 'Public Health', 'prereqs': 'MATH 231, MATH 232, MATH 233, STOR 155', 'prior_exp': '1-2 years', 'AP_Principles': 'No', 'AP_A': 'No', 'other_comp': 'UNC', 'prior_time': '1 month or so', 'languages': 'Python, R / Matlab / SAS, Other', 'residency': 'Out-of-state', 'on_campus': 'Yes', 'international': 'I am living in the United States', 'section': 'Section 2 - 5:00pm - Async', 'lesson_time': '1', 'sync_perf': '3', 'all_sync': '3', 'own_notes': '7', 'own_examples': '4', 'oh_visits': '2', 'ls_effective': '3', 'lsqs_effective': '5', 'programming_effective': '3', 'qz_effective': '4', 'oh_effective': '7.0', 'tutoring_effective': '3.0', 'kaki_effective': '4.0', 'pace': '5', 'difficulty': '5', 'understanding': '5', 'interested': '7', 'valuable': '7', 'would_recommend': '6'}
{'': '1', 'row_number': '1', 'year': '23', 'unc_status': 'Returning UNC Student', 'comp_major': 'No', 'prim

Next, I am going to select the `comp_major` column of the survey.csv file. Here are the first five rows:

In [78]:
from data_utils import head, select
subset: dict[str, list[str]] = select(columns, ['comp_major', 'row_number'])
head(subset, 5)

{'comp_major': ['No', 'No', 'No', 'No', 'Yes - BS'],
 'row_number': ['0', '1', '2', '3', '4']}

Next, I am going to figure out how many COMP BS majors took COMP 110 this past spring.

In [81]:
from data_utils import count
maj_counts = count(subset['comp_major'])
rows = list(subset['row_number'])

print(f"The total amount of kids in COMP 110 this past spring was {rows[-1]}, and out of that amount, {counts['Yes - BS']} are COMP BS majors.")

The total amount of kids in COMP 110 this past spring was 508, and out of that amount, 60 are COMP BS majors.


* Total Amount of `Students in COMP 110 S2021`: 508
* Total Amount of `COMP BS majors`: 60

# ` In Conclusion: `
* Out of the 508 students at UNC taking COMP 110 this past spring, 60 of these students are Computer Science (Bachelor of Science) Majors. I have pondered this question before about my own session as I am in search of COMP BS majors to collaborate on topics with and learn from. With UNC being a mainly liberal arts college, sometimes it can feel lonely being a COMP SCI BS major, and now I feel less alone with these statistics!